Sources/Sentry/include/SentryPermissionsObserver.h

```diff
@@ -0,0 +1,19 @@
+#import "SentryDefines.h"
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef enum {
+    kPermissionStatusUnknown,
+    kPermissionStatusGranted,
+    kPermissionStatusDenied
+} SentryPermissionStatus;
+
+@interface SentryPermissionsObserver : NSObject
+
+@property (nonatomic) SentryPermissionStatus hasPushPermission;
```

Comment:
A boolean can't be nullable, wrapping it in an NSNumber felt wrong, so I introduced `SentryPermissionStatus` instead.